osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -105,20 +105,17 @@ Status FirehoseLogForwarder::setUp() {
   }
 
   // Make sure we can connect to designated stream
-  Aws::Firehose::Model::ListDeliveryStreamsRequest r;
-  auto result = client_->ListDeliveryStreams(r).GetResult();
-  std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
-
-  if (std::find(stream_names.begin(),
-                stream_names.end(),
-                FLAGS_aws_firehose_stream) == stream_names.end()) {
-    return Status(1,
-                  "Could not find Firehose delivery stream: " +
-                      FLAGS_aws_firehose_stream);
+  Aws::Firehose::Model::DescribeDeliveryStreamRequest r;
+  r.SetDeliveryStreamName(FLAGS_aws_firehose_stream);
+  auto result = client_->DescribeDeliveryStream(r).GetResult();
+  auto description = result.GetDeliveryStreamDescription();
+  auto stream_name = description.GetDeliveryStreamName();
+  if (stream_name.empty()){
```

Comment:
@marpaia thx for the tips.  Ill take a look and see if I can add some short circuits earlier in the chain.  I was mostly just following the code that was already there, but I see what you mean.

i.e.  the previous code wasnt checking for null or error before calling ````GetResult```` and ````GetDeliveryStreamNames````
````
Aws::Firehose::Model::ListDeliveryStreamsRequest r;
auto result = client_->ListDeliveryStreams(r).GetResult();
std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
````